test/kubernetes/e2e/features/route_options/route_options.go

```diff
@@ -0,0 +1,129 @@
+package route_options
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/gloo/pkg/utils/kubeutils/kubectl"
+	"github.com/solo-io/gloo/pkg/utils/requestutils/curl"
+	"github.com/solo-io/gloo/test/kubernetes/e2e"
+	"github.com/solo-io/gloo/test/kubernetes/testutils/assertions"
+	"github.com/solo-io/gloo/test/kubernetes/testutils/operations"
+	"github.com/solo-io/go-utils/threadsafe"
+	"github.com/solo-io/skv2/codegen/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	targetRefManifest      = filepath.Join(util.MustGetThisDir(), "inputs/fault-injection-targetref.yaml")
+	filterExtensioManifest = filepath.Join(util.MustGetThisDir(), "inputs/fault-injection-filter-extension.yaml")
+
+	// When we apply the deployer-provision.yaml file, we expect resources to be created with this metadata
+	glooProxyObjectMeta = metav1.ObjectMeta{
+		Name:      "gloo-proxy-gw",
+		Namespace: "default",
+	}
+	proxyDeployment = &appsv1.Deployment{ObjectMeta: glooProxyObjectMeta}
+	proxyService    = &corev1.Service{ObjectMeta: glooProxyObjectMeta}
+
+	curlPod = &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "curl",
+			Namespace: "curl",
+		},
+	}
+
+	// RouteOption resource to be created
+	routeOptionMeta = metav1.ObjectMeta{
+		Name:      "teapot-fault-injection",
+		Namespace: "default",
+	}
+)
+
+var ConfigureRouteOptionsWithTargetRef = e2e.Test{
+	Name:        "RouteOptions.ConfigureRouteOptionsWithTargetRef",
+	Description: "the RouteOptions will configure fault inject with a targetRef",
+	Test: func(ctx context.Context, installation *e2e.TestInstallation) {
+		trafficRefRoutingOp := operations.ReversibleOperation{
+			Do: &operations.BasicOperation{
+				OpName:   fmt.Sprintf("apply-manifest-%s", filepath.Base(targetRefManifest)),
+				OpAction: installation.Actions.Kubectl().NewApplyManifestAction(targetRefManifest),
+				OpAssertion: func(ctx context.Context) {
+					// First check resources are created for Gateay
+					installation.Assertions.ObjectsExist(proxyService, proxyDeployment)
+
+					// Check fault injection is applied
+					CheckFaultInjectionFromCluster()(ctx)
+
+					// TODO(npolshak) Check status on solo-apis client object once route option status support is added
+				},
```

Comment:
I don't think this is doing exactly what you may want. It seems like you want to define a list of assertions. Right now you are creating one assertion, and that calls a function to create the objects assert function, but doesn't execute it. I think what you want is
```suggestion
				OpAssertions: []assertions.ClusterAssertion {
					// First check resources are created for Gateay
					installation.Assertions.ObjectsExist(proxyService, proxyDeployment),

					// Check fault injection is applied
					CheckFaultInjectionFromCluster(),

					// TODO(npolshak) Check status on solo-apis client object once route option status support is added
				},
```